src/models/mint.ts

```diff
@@ -1,85 +1,37 @@
-}
-
-export interface ERC721MintData {
-  /**
-   * Address to transfer the NFT after minting,
-   * And this address will be considered as the Owner.
-   */
-  to: string;
-  /**
-   * This is the suffix for the tokenURI, typically our prefix would be "ipfs://".
-   */
-  uri: string;
-  /**
-   * creators is supplied as an array of Owner, this array should contain all
-   * the addresses (with their respective part of the creation - in basis points)
-   * who are considered the authors/creators of this token.
-   * The address array is public and can be queried by anyone.
-   * Sum of fields value in this array should be 10000 (100% in basis points)
-   */
-  creators: PartOwner[];
-  /**
-   * royalties are an array of addresses and values.
-   * The fees array is public and can be queried by anyone.
-   * Values are specified in basis points. So for example, 200 means 2%
-   */
-  royalties: PartOwner[];
+export interface MintData {
   /**
-   * Signatures is an array of wallet signatures for this transaction from every creator,
-   * the only exception to this is when the creator sends the mint transaction - then empty signature
-   * can be passed for the executing transaction creator.
-   */
-  signatures: string[];
-}
+   *
+   *  This is the bare minimum info needed to create an NFT. The SDK should handle assigning
+   *  tokenID, pushing metadata to IPFS, etc.
+   *
+   **/
 
-export interface ERC1155MintData extends ERC721MintData {
-  /**
-   * supply should be supplied as an uint256, this is the number of copies (Editions)
-   * of this token that exist. (Maximum value is 2**256 - 1).
-   */
-  supply: BigNumberish;
-}
+  // Defaults to signer address if single creator and `to` not speficied
+  to?: string;
 
-export type MintData = ERC721MintData | ERC1155MintData;
+  // Defaults to [signer] if not specified.
+  creators?: PartOwner[];
 
-export interface ERC721LazyMint extends Omit<ERC721MintData, "to"> {
-  /**
-   * NFT as ERC-721 Non-Fungible Token Standard.
-   */
-  "@type": "ERC721";
+  // Signatures will be required for multi-creator NFTs.
+  // Defaults to [] if not specified.
+  signatures?: Signature[];
 
-  contract: string;
-}
+  // Defaults to [] if not specified.
+  royalties: PartOwner[];
 
-export interface ERC1155LazyMint extends Omit<ERC1155MintData, "to"> {
-  /**
-   * NFT as ERC-1155 Multi Token Standard.
-   */
-  "@type": "ERC1155";
+  // Contract type (ERC721 vs 1155) is inferred by presence of supply field with value > 1.
+  // Defaults to 1.
+  supply?: BigNumberish;
 
-  contract: string;
+  // Required metadata.
+  metadata: MintMetadata;
 }
 
-export type LazyMintData = ERC721LazyMint | ERC1155LazyMint;
-
-export interface LazyMintResponse {
-  id: string;
-  contract: string;
-  tokenId: string;
-  creator?: string;
-  supply: number;
-  lazySupply: number;
-  owners: string[];
-  royalties: PartOwner[];
-  pending: ItemTransfer[];
+export interface Signature {
```

Comment:
Do we need an object that only has a single property? Or can it just be a `string` named `signature`?